src/ApiClient.php

```diff
@@ -34,38 +33,30 @@
  * It handles requesting and setting up what is necessary to execute calls to the endpoints.
  *
  * @see ApiClientInterface
- * @see TokenAwareTrait
+ * @see ConfigurationAwareTrait
  */
 class ApiClient implements ApiClientInterface
 {
-    /**
-     * @var array<string, mixed>
-     */
-    protected array $configuration = [];
+    use ConfigurationAwareTrait;
```

Comment:
I noticed that both endpoints and the client are using the same structures for storing the config. I moved it to the trait.